lib/jekyll/polyglot/patches/jekyll/site.rb

```diff
@@ -89,12 +96,17 @@ def process_active_language
     end
 
     # assigns natural permalinks to documents and prioritizes documents with
-    # active_lang languages over others
+    # active_lang languages over others.  If lang can be derived from the
+    # source path, that is used, otherwise lang must be set in front matter.
     def coordinate_documents(docs)
       regex = document_url_regex
       approved = {}
       docs.each do |doc|
-        lang = doc.data['lang'] || @default_lang
+        segments = doc.relative_path.split('/')
+        if doc.relative_path[0] == '_' and segments.length > 2
+        then lang = doc.relative_path.split('/')[1]
+        else lang = doc.data['lang'] || @default_lang
+        end
```

Comment:
I know of polyglot usage that would break with this `If lang can be derived from the
    # source path, that is used, otherwise lang must be set in front matter.` functionality. While convenient, not everyone has their layout setup this way, and posts in a directory like:

```
_posts/salads/2010-03-01-salad-recipes-en.md
_posts/entrees/2010-03-01-entrees-sv.md
_posts/happy-hour/2010-03-01-happy-hour-fr.md
```

will mean expected langs `en` `sv` `fr` are confused for directories `salads` `entrees` `happy hour`

Reply:
other changes seem fine, but I'd ask that you back this out

Reply:
Ok, I can either maintain it in our fork, or make a config option.  Any preference?

Reply:
Oh wait, now that I look at this again, I had a local change that didn't make it here that made sure that the path element fit the locale pattern, otherwise it was not treated like `lang:`.  In any case, I think I'll make this feature enabled by a config option, since it is quite useful if anyone is using [po4a](https://po4a.org) to translate their jekyll/polyglot site.